src/pkg/packager/actions.go

```diff
@@ -0,0 +1,165 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2021-Present The Zarf Authors
+
+// Package packager contains functions for interacting with, managing and deploying zarf packages
+package packager
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"regexp"
+	"runtime"
+	"strings"
+	"time"
+
+	"github.com/defenseunicorns/zarf/src/pkg/message"
+	"github.com/defenseunicorns/zarf/src/pkg/utils/exec"
+	"github.com/defenseunicorns/zarf/src/types"
+)
+
+// Run commands that a component has provided
+func (p *Packager) runComponentActions(actionSet types.ZarfComponentActionSet, actions []types.ZarfComponentAction) error {
+ACTION:
+	for _, a := range actions {
+		spinner := message.NewProgressSpinner("Running command \"%s\"", a.Cmd)
+		defer spinner.Success()
+
+		var (
+			ctx    context.Context
+			cancel context.CancelFunc
+			cmd    string
+			err    error
+		)
+
+		cfg := actionGetCfg(actionSet, a)
+
+		if cmd, err = actionCmdMutation(a.Cmd); err != nil {
+			spinner.Errorf(err, "Error mutating command: %s", cmd)
+		}
+
+		// Otherwise, run the command with a timeout handler
+		spinner.Updatef("Waiting for command \"%s\" (timeout: %d seconds)", cmd, cfg.MaxTotalSeconds)
+
+		duration := time.Duration(cfg.MaxTotalSeconds) * time.Second
+		timeout := time.After(duration)
+
+		// Keep trying until the max retries is reached
+		for remaining := cfg.MaxRetries + 1; remaining > 0; remaining-- {
```

Comment:
This will add 1 to the maxRetries giving one more than you actually expect.  If you wanted 0 to be equal to one then you would need to do something like this:

```suggestion
		for remaining := cfg.MaxRetries == 0 ? 1 : cfg.MaxRetries; remaining > 0; remaining-- {
```

Reply:
NVM on this.  I was thinking about the word "retries" wrong...